.github/workflows/cron-update-new-year-copyright.yml

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+name: Update Copyright Year
+description: Update the year on copyright notice with a scheduled run on Jan 1st every year
+
+on:
+  schedule:
+    - cron: "0 0 1 1 *"  # Run once a year on January 1st
+  workflow_dispatch:
+
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-year:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: true
```

Comment:
note for reviewers

true is required here so that the `git push` actions below can use these credentials

Reply:
Ah, seems like apache wants this to be false always: `if using the 'checkout' action, always enable persist-credentials: false ` from https://cwiki.apache.org/confluence/display/BUILDS/GitHub+Actions+Security

I will fix it now.

Reply:
This is fixed now. Using false.